Mood history

jamesMckinnonSchool opened the pull request:
Resolves: #130
Resolves: #145
Resolves: #146

**Description**

- Added `MoodHistory` and `MoodHistoryAdapter` classes.
- Added a mood history page.
- Created a search bar and sorted moods on page.
- Updated navigation to mood history for profile page.
- Added navigation for mood details page from a mood event in mood history list.

**Reviewer Checklist**

- [ ] Are all relevant issues referenced by this PR?
- [ ] Does the app still build?
- [ ] Do all tests pass? (TODO - after demo)
- [ ] Does the feature work as expected?
- [ ] Does the feature still keep the app bug free?
- [ ] Is the description of the changes correct/present?

---

jamesMckinnonSchool committed:
Mood history page with back button

jamesMckinnonSchool committed:
Displayed fake mood events in list in mood history page and added search bar.

jamesMckinnonSchool committed:
added search bar and icons, and styled mood history page

jamesMckinnonSchool committed:
Merge branch 'main' into mood_history

jamesMckinnonSchool committed:
renamed mood event layout for mood history page

jamesMckinnonSchool committed:
Merge branch 'main' into mood_history

jamesMckinnonSchool committed:
Updated mood history adapter, sorted mood history items by default

jamesMckinnonSchool committed:
UI tests for mood history page

jamesMckinnonSchool committed:
updated firebase stuff

# Conflicts:
#	code/app/src/main/java/com/kernelcrew/moodapp/ui/AuthSignIn.java

jamesMckinnonSchool committed:
updated firebase stuff

jamesMckinnonSchool committed:
updated firebase stuff

# Conflicts:
#	code/app/src/main/java/com/kernelcrew/moodapp/ui/AuthSignIn.java

jamesMckinnonSchool committed:
Merge branch 'main' into mood_history

jamesMckinnonSchool committed:
made mood history button on profile functional

jamesMckinnonSchool committed:
dynamic moodHistory back button destination based on initial location

jamesMckinnonSchool committed:
Moods come from DB now for Mood History

jamesMckinnonSchool committed:
clicking mood event on mood history page brings user to mood details

jamesMckinnonSchool committed:
before merge

jamesMckinnonSchool committed:
merged main

jamesMckinnonSchool committed:
updated mood history to account for new MoodEvent class

jamesMckinnonSchool committed:
Merge branch 'main' into mood_history

TahaKamil commented on .firebaserc:
This should be in your .gitignore?

TahaKamil commented on .DS_Store:
Also add this to .gitignore

TahaKamil commented on code/app/src/main/res/values/themes.xml:
I don't know if we want to use themes. This would be a point of discussion, because some UI might rely on default themes.

TahaKamil commented on code/app/src/androidTest/java/com/kernelcrew/moodapp/MoodHistoryTests.java:
```diff
@@ -0,0 +1,114 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
```
I don't think you added the dependencies into the gradle files, for the lab it was already in the gradle!

TahaKamil reviewed (dismissed):
Just took a quick look at code/diff and didn't do any testing of UI. The fixes I asked should be really easy to implement. Good luck 🫡

jamesMckinnonSchool committed:
unit tests

jamesMckinnonSchool commented:
I don't know what's causing the error when running UI tests yet. It was happening on my computer for my tests tests so I deleted those tests to try to get a pull request completed, then now it's happening for all of the existing tests that other people made:

com.kernelcrew.moodapp.CreateMoodEventTest > createNewMoodReasonJustEnoughWords[emulator-5554 - 14] FAILED 
	java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/test/platform/concurrent/DirectExecutor;
	at androidx.test.espresso.InteractionResultsHandler.gatherAnyResult(InteractionResultsHandler.java:52)

PossiblyAShrub committed:
Remove extra files

PossiblyAShrub committed:
Fix deps issue leading to CI error

PossiblyAShrub committed:
Remove .DS_Store

PossiblyAShrub committed:
Format the build.gradle

PossiblyAShrub committed:
Fix unit tests

PossiblyAShrub committed:
Merge branch 'main' into mood_history

PossiblyAShrub committed:
Rename Controller to Provider

PossiblyAShrub committed:
Fix lint error

PossiblyAShrub commented:
@jamesMckinnonSchool This should be good to continue work on. I'm happy to merge right away if you'd like. Although, it looks like we still need to actually filter by uid in the mood history page itself.

jamesMckinnonSchool commented:
@PossiblyAShrub Okay sounds good, if you could merge it that would be great. Thanks for getting that sorted out. 

For the filtering, as far as I know the mood history should be just the user's mood history and then other users' data is in the home feed. I could be wrong but at least for the Project Part 3 Milestone requirements it should be only the current user's mood events in mood history.

jamesMckinnonSchool commented:
@PossiblyAShrub Oh I see what you're saying now. All user data is in a common MoodEvents collection and I'll need to exclusively pull the current user's mood events.

jamesMckinnonSchool committed:
Only current user's mood events in mood history now. Fixed issue with list not updating after navigating.

# Conflicts:
#	code/app/src/main/java/com/kernelcrew/moodapp/data/MoodEventProvider.java
#	code/app/src/main/java/com/kernelcrew/moodapp/ui/MoodHistory.java

jamesMckinnonSchool committed:
removed reference to MoodEventController
